src/shortest-path/astar.js

```diff
@@ -0,0 +1,172 @@
+/**
+ * Graphology A* Shortest Path
+ * ==================================
+ *
+ * Graphology implementation of A* shortest path for weighted graphs.
+ */
+var isGraph = require('graphology-utils/is-graph');
+var createEdgeWeightGetter =
+  require('graphology-utils/getters').createEdgeWeightGetter;
+var Heap = require('mnemonist/heap');
+
+/**
+ * Defaults & helpers.
+ */
+var DEFAULT_WEIGHT_ATTRIBUTE = 'weight';
+
+function ASTAR_HEAP_COMPARATOR(a, b) {
+  if (a[0] > b[0]) return 1;
```

Comment:
tbh, I'm not 100% sure. I copied this from networkx and they just had the standard heap pop/push operators, I think. Here's their code: https://github.com/networkx/networkx/blob/main/networkx/algorithms/shortest_paths/astar.py#L103-L104

In the A* algorithm, the highest proprity should be the element that has the lowest weight + the value of the heuristic. In the way it's implemented here, that's the first element of the array. However, I don't know if there should be a "backup" priority. Maybe just the weight 🤷‍♂️

Reply:
The python code you show will consider the ordering of the tuples inserted into the heap array. This means comparison will be done on the first item, then the second, then the first etc. in case of ties. So I think you will need to refine your comparator to reflect this. What's more, the final item is often the node key or some node index to break ties deterministically.

In networkx code, the heap ordering seems to only happen on some cost, then an incremental id to break ties: https://github.com/networkx/networkx/blob/main/networkx/algorithms/shortest_paths/astar.py#L115, which means that items 2, 3 and 4 are not used for sorting indeed.

In our case, this probably mean you must update the comparator to use the second element of the item to break up ties.